cow_instrument/DetectorInfo.h

```diff
@@ -40,27 +40,45 @@ template <typename InstTree> class DetectorInfo {
 
   bool isMonitor(size_t detectorIndex) const;
 
-  void initL2();
+  double l2(size_t detectorIndex) const;
 
-  void initL1();
+  Eigen::Vector3d position(size_t componentIndex) const;
 
-  double l2(size_t detectorIndex) const;
+  Eigen::Vector3d positionDetector(size_t detectorIndex) const;
```

Comment:
Yes I had considered that this could be a problem. Particularly the lack of type safety around the different index types. Now that all `Components` including non-detectors  have been refactored out of the prototype, we still need a way to get information on any physical part of the instrument (hence the component index). 

After making detectors simple arrays of positions etc, my profiler indicated that the `l1` & `l2` calculations continued to be expensive due to the visualisation that still existed.

Im open to suggestions. One option would be to make the indexes `Int2Type` value objects. But for the time being I've at least tried to make the naming consistent.